pkg/services/k8s/resources/dashboard/watcher.go

```diff
@@ -21,29 +22,45 @@ type watcher struct {
 	enabled        bool
 	log            log.Logger
 	dashboardStore database.DashboardSQLStore
+	folders        folder.FolderStore
 }
 
 func ProvideWatcher(
 	features featuremgmt.FeatureToggles,
 	dashboardStore database.DashboardSQLStore,
 	userService user.Service,
 	accessControlService accesscontrol.Service,
+	folders folder.FolderStore,
 ) (*watcher, error) {
 	c := watcher{
 		enabled:        features.IsEnabled(featuremgmt.FlagK8S),
 		log:            log.New("k8s.dashboards.controller"),
 		dashboardStore: dashboardStore,
+		folders:        folders,
 	}
 	return &c, nil
 }
 
-func (c *watcher) Add(ctx context.Context, dash *Dashboard) error {
-	c.log.Debug("adding dashboard", "dash", dash)
-
-	js, _ := json.MarshalIndent(dash, "", "  ")
-	fmt.Printf("-------- WATCHER ---------")
-	fmt.Printf("%s", string(js))
+// UID is currently saved in the dashboard body and the name may be a hash
+func getValidUID(dash *Dashboard) (string, error) {
+	uid := ""
+	if dash.Spec.Uid != nil {
+		uid = *dash.Spec.Uid
+	}
+	if uid == "" {
+		uid = dash.GetName()
+	} else if dash.GetName() != GrafanaUIDToK8sName(uid) {
+		return uid, fmt.Errorf("UID and k8s name do not match")
+	}
+	return uid, nil
+}
 
+func (c *watcher) Add(ctx context.Context, dash *Dashboard) error {
+	uid, err := getValidUID(dash)
```

Comment:
do we still need to check if the dashboard exists before attempting to create again? i'm seeing a unique constraint error on startup